search volunteer ops via keywords

Maroo-b opened the pull request:
WIP PR for https://www.pivotaltracker.com/n/projects/742821/stories/74774180

tansaku commented:
@Maroo-b want to try and deal with the hound comments?  If you're using RubyMine there's a hot-key short cut to fix these things (alt-cmd-l on OSX)

tansaku commented on app/controllers/volunteer_ops_controller.rb:
```diff
@@ -1,7 +1,15 @@
+  before_action :authorize, :except => [:search, :show, :index]
 
+  def search
+    @key_words = params[:q]
```
maybe @keywords?  or perhaps better call this @query ...? - arguably I think we are on full text search here ...

Maroo-b replied:
ok I'll edit the variable name to @query

Maroo-b commented:
@tansaku at first hound informed me that line was too long so I split it and after the changes it wants the expression to be on it's own line :/

tansaku commented:
yeah, hound's a mixed bag - I'm not sure how much value we're getting from it ... which editor are you using BTW?

Maroo-b commented:
@tansaku I'm using VIM + Tmux combination

tansaku commented:
oh cool - I've got some minor edits on this branch that might be faster for me to make - do you want to give me commit access to https://github.com/Maroo-b/LocalSupport do I can push them over to you?

Maroo-b commented:
@tansaku  I added you to the repo can you try now ?

tansaku commented:
will do - give me a moment - just trying to configure some grader notifications

tansaku commented on config/routes.rb:
```diff
@@ -16,7 +16,9 @@
   resources :pages, only: [:index, :new, :create, :edit]
-  resources :volunteer_ops, :only => [:index, :edit, :show, :update, :destroy]
+  resources :volunteer_ops, :only => [:index, :edit, :show, :update, :destroy] do
+    get 'search', on: :collection
```
what's the advantage of this?  we don't have it for our organisation search at the moment ...

Maroo-b replied:
@tansaku for the organisation search currently we are using https://github.com/AgileVentures/LocalSupport/blob/develop/config/routes.rb#L6 , normally the route accepts only **GET** 
The advantage of using the highlighted syntax is to regroup routes by resources instead of adding routes separately.
I suggested that we can refactor some routes like instead of using:
`resources :volunteer_ops, :only => [:index, :edit, :show, :update, :destroy]` we can use `resources :volunteer_ops, :except => [:new, :create]`

tansaku replied:
if that's just reorganizing how they appear in rake routes then that sounds reasonable - want to open a separate ticket for refactoring that?   If so, please be sure to include copies of before and after on the rake routes (and note any other side effects)

tansaku commented on app/views/volunteer_ops/index.html.erb:
```diff
@@ -1,3 +1,14 @@
+        <%= label_tag(:q, "Search Text", class: 'field-label') %>
+      </div>
+      <div class="right-side">
+        <%= text_field_tag(:q, @query) %>
```
note that we have some untested functionality here, i.e. that the query string is visible after the each search

tansaku commented:
@Maroo-b have a look at some of the changes I made.

Also note I switched the text from "Optional search text" to "Search text", since we don't have categories yet, so it's not optional ... however I think that means we might need a little CSS work:

![interface view with "Search text" leaving a gap](https://www.dropbox.com/s/ufuv1w4yvfihi1e/Screenshot%202016-03-22%2012.45.07.png?dl=1)

Keywords commented:
Hello,

I don’t know why I’m included in your communications, could you please cancel me (mdesanctis@keywordsintl.com) from this thread?

Cheers
Marco

Marco De Sanctis
Rome Office Manager

________________________________
Binari Sonori s.r.l. a Keywords studio
Your Sound Production and Localization Partner
Via Tiberio Imperatore 15, 00145 Roma - Italia
Phone: +39-06-44202521- +39 340 5369085
Web: www.binarisonori.com<http://www.binarisonori.com/> - www.keywordsintl.com<http://www.keywordsintl.com/>
________________________________

From: Sam Joseph [mailto:notifications@github.com]
Sent: Tuesday, March 22, 2016 1:47 PM
To: AgileVentures/LocalSupport
Cc: Keywords
Subject: Re: [LocalSupport] search volunteer ops via keywords [#74774180] (#272)


@Maroo-b<https://github.com/Maroo-b> have a look at some of the changes I made.

Also note I switched the text from "Optional search text" to "Search text", since we don't have categories yet, so it's not optional ... however I think that means we might need a little CSS work:

[Immagine rimossa dal mittente. interface view with "Search text" leaving a gap]<https://camo.githubusercontent.com/0860d35c2b0a50707abee24f01b5687ed7ef3954/68747470733a2f2f7777772e64726f70626f782e636f6d2f732f7566757631773479766669686931652f53637265656e73686f74253230323031362d30332d323225323031322e34352e30372e706e673f646c3d31>

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/AgileVentures/LocalSupport/pull/272#issuecomment-199795287>

tansaku commented:
Sincere apologies Marco - I have no idea why you are being pulled in - the only people who should be messaged are me (tansaku@gmail.com) and Maroo who submitted it ... is this a Github error?  Maroo - what email are you using on github?

tansaku commented:
Marco, are you http://github.com/maroo ?

Keywords commented:
I don’t know if I’m maroo , I reca only ll I’ve been added ages ago due to a project with SongHi… but since then I’ve never received an email but these last few days
Cheers
Marco

Marco De Sanctis
Rome Office Manager

________________________________
Binari Sonori s.r.l. a Keywords studio
Your Sound Production and Localization Partner
Via Tiberio Imperatore 15, 00145 Roma - Italia
Phone: +39-06-44202521- +39 340 5369085
Web: www.binarisonori.com<http://www.binarisonori.com/> - www.keywordsintl.com<http://www.keywordsintl.com/>
________________________________

From: Sam Joseph [mailto:notifications@github.com]
Sent: Tuesday, March 22, 2016 2:11 PM
To: AgileVentures/LocalSupport
Cc: Keywords
Subject: Re: [LocalSupport] search volunteer ops via keywords [#74774180] (#272)


Marco, are you http://github.com/maroo<https://github.com/maroo> ?

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/AgileVentures/LocalSupport/pull/272#issuecomment-199805915>

Maroo-b commented:
@tansaku the primary email I'm using is marouen.it.technology@gmail.com, I guess the user 'Marco' was added to discussion when we discussed changed variable name of **keywords** to **query**

tansaku commented:
I think it might be a github bug - they do notifications against things like @tansaku - which notifies me, but Maroo you've got a hypen in your username which the git notification system maybe can't handle ... I'll file a bug ticket with GitHub

tansaku commented:
response from github indicates the username for Marco was either keywords or query - looks like we have to be careful talking about ruby instance variables outside of a code block!

Maroo-b commented:
@tansaku I added test to check if the query string is visible after  each search. Also I updated the search form 

![search_form](https://cloud.githubusercontent.com/assets/7128972/13958338/fff52c66-f04f-11e5-86a5-de9ebff62842.png)


If everything is ok should I squash the commits ?

tansaku commented:
squashing would be great thanks @Maroo-b

Maroo-b commented:
@tansaku Squash done.

Maroo-b commented:
@tansaku I updated the form styling again to optimize mobile view.

tansaku commented:
thanks @Maroo-b will try to review soon :-)

tansaku commented:
@Maroo-b reviewing this I think we need a different CSS style associated with the search text.  Take a look at this screen shot:

![](https://www.dropbox.com/s/zru7dkihn0bceg7/Screenshot%202016-03-24%2010.16.20.png?dl=1)

At the moment the "Search Text" label is in a form that has the id "organisation_search".  I think this should probably be updated to "volunteer_ops_search" so that we can then associated a different style (and text width), so that we have the text entry field fill up more of the width and reduce the amount of blank space (and make it clear that the label is associated with the text box).

![](https://www.dropbox.com/s/8ualo9da91el314/Screenshot%202016-03-24%2010.19.34.png?dl=1)

I just dynamically set to 110px which I think looks a bit better ...

Maroo-b commented:
@tansaku it seems that you didn't pull the new changes, (I used push -f to override to previous commits)

tansaku commented:
ah yes, sorry - right I've got that locally now:

![](https://www.dropbox.com/s/ukgcvosu1l5hf2v/Screenshot%202016-03-24%2010.25.46.png?dl=1)

That's an improvement, but I'm concerned that the search button now doesn't aline with the box below, and that the curves on the end of the text field are now different from those on the organisation search:

![](https://www.dropbox.com/s/83yyfdyd82hgo0t/Screenshot%202016-03-24%2010.26.55.png?dl=1)

I think it's important that we have a consistent look and feel across the different parts of the site.  Would you consider rolling back and then just making the smaller change in the way I suggested above?

Maroo-b commented:
@tansaku ok I'll edit the form style as suggested.

Keywords commented:
Hello Guys,

I’m still in CCed in all your message. Could you cancel me or tell me how to do it?

Cheers
M

Marco De Sanctis
Rome Office Manager

________________________________
Binari Sonori s.r.l. a Keywords studio
Your Sound Production and Localization Partner
Via Tiberio Imperatore 15, 00145 Roma - Italia
Phone: +39-06-44202521- +39 340 5369085
Web: www.binarisonori.com<http://www.binarisonori.com/> - www.keywordsintl.com<http://www.keywordsintl.com/>
________________________________

From: Maroo-b [mailto:notifications@github.com]
Sent: Thursday, March 24, 2016 11:32 AM
To: AgileVentures/LocalSupport
Cc: Keywords
Subject: Re: [LocalSupport] search volunteer ops via keywords [#74774180] (#272)


@tansaku<https://github.com/tansaku> ok I'll edit the form style as suggested.

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/AgileVentures/LocalSupport/pull/272#issuecomment-200778114>

Maroo-b commented:
@Keywords no problem you can unsubscribe to the notification by clicking on **Unsubscribe** button from side menu

![selection_002](https://cloud.githubusercontent.com/assets/7128972/14014988/41d3108c-f1b6-11e5-8297-5bd42b68a646.png)

Maroo-b commented:
@tansaku I edited the form style as required. Once approved I'll squash the commits.

tansaku commented on app/assets/stylesheets/bootstrap_and_overrides.css.less:
```diff
@@ -58,3 +58,12 @@ h3 {
   color: #185000;
 }
+
+.form-search {
```
is this redundant now?

Maroo-b replied:
I forgot to remove it.

tansaku commented:
we might want that left-side width at 120px or a tiny bit of left padding or margin on the search text if you have a moment:

![](https://www.dropbox.com/s/wrey8npaddi1x4p/Screenshot%202016-03-24%2013.15.52.png?dl=1)

but otherwise we're there - fix that - squash the commits and we'll get this merged in 

Great work!

Maroo-b commented:
@tansaku changed made and commits squashed, it seems that I tested only using chrome browser.

tansaku commented:
@Maroo-b thanks - looking great - we might just want to review the hound comments on the CSS - I know you just followed the logic from some of our existing CSS so don't feel compelled to do that yourself - I might have a go myself and push later today - let's see
